src/main/java/org/apache/commons/io/monitor/FileAlterationMonitor.java

```diff
@@ -165,8 +165,8 @@ public synchronized void stop(final long stopInterval) throws Exception {
         running = false;
         try {
             thread.join(stopInterval);
+            thread.interrupt();
         } catch (final InterruptedException e) {
-            Thread.currentThread().interrupt();
```

Comment:
Is swallowing an `InterruptedException` a really bad practice (at least according to `Java Concurrency in Practice` and https://www.ibm.com/developerworks/library/j-jtp05236/)?

Reply:
Regardless of the implementation of this fix, it would be great to provide a unit test that fails without the main patch. Otherwise, "confidence is low" and regression will happen.

Reply:
Hi @garydgregory .
@SvetlinZarev was faster than i was: https://github.com/apache/commons-io/pull/58